File: API/Entities/Order.cs
namespace API.Entities;

public class Order
{
    public int Id { get; set; }
    public int UserId { get; set; }
    public User User { get; set; }
    public double Total { get; set; }
    public DateTime CreatedAt { get; set; }
    public OrderAddress OrderAddress { get; set; }
    public ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
}

File: Client/src/types/order.ts
import { User } from './authResponse.ts';
import { AddressDetails } from './addressDetails.ts';

export type Order = {
    id: number;
    userId: number;
    user: User;
    total: number;
    createdAt: string;
    orderAddress: Omit<AddressDetails, 'UserAddressId'>;
    orderItems: OrderItem[];
};

export type OrderItem = {
    id: number;
    orderId: number;
    total: number;
    quantity: number;
    productId: number | null;
};
